corehq/apps/userreports/dbaccessors.py

```diff
@@ -65,3 +70,30 @@ def get_all_report_configs():
 def delete_all_report_configs():
     from corehq.apps.userreports.models import ReportConfiguration
     delete_all_docs_by_doc_type(ReportConfiguration.get_db(), ('ReportConfiguration',))
+
+
+def get_all_table_names_for_domain(domain):
+    engine = connection_manager.get_engine(UCR_ENGINE_ID)
+    with engine.begin() as connection:
+        possible_matches = connection.execute("select tablename from pg_tables where tablename ~ %s", domain)
+    # DO NOT assume just because the domain is in the table name
+    # that the table is in the domain
+    all_table_names_for_domain = []
+    for table_name, in possible_matches:
+        match = re.match(r'^(config_report|ucr)_(.*)_([0-9a-f]{0,29})_([0-9a-f]{8})$', table_name)
+        if match:
+            prefix, table_domain, table_big_hash, table_little_hash = match.groups()
+            if prefix == 'ucr':
+                table_name_check = get_table_name(domain, table_big_hash)
+            else:
+                table_name_check = get_legacy_table_name(domain, table_big_hash)
+            if table_name == table_name_check:
```

Comment:
This would only work in cases where the table name hasn't been trimmed:
```
In [5]: get_table_name('domain', 'very-long-table-name-that-will-get-trimmed')
Out[5]: 'ucr_domain_very-long-table-name-that-will_8540a3df'

In [6]: get_table_name('domain', 'very-long-table-name-that-will')
Out[6]: 'ucr_domain_very-long-table-name-that-will_547c90e1'
```